### PR TITLE
Added: A method to get a Page by ID from the Document, a method to get a Page's original from its copy

### PR DESCRIPTION
**Pull request by iftwigs**
Added methods to retrieve an original Page from its copy and to get a Page by its ID from the Document; relaxed the conditions of setting Category – now a Document's Category will always be NO_CATEGORY when specified as None.

### CODE REVIEW

**Review by zypriafl — DISMISSED** [review]

File: tests/trainer/test_file_splitting.py [hunk 1]
```diff
@@ -209,27 +214,26 @@ def test_splitting_no_category_document(self):
...
-        test_document = self.file_splitting_model.tokenizer.tokenize(deepcopy(self.test_document))
-        pred = splitting_ai.propose_split_documents(test_document, return_pages=True)[0]
+        pred = splitting_ai.propose_split_documents(self.test_document, return_pages=True)[0]
         for page in pred.pages():
             if page.number in (1, 3, 5):
                 assert page.is_first_page
             else:
                 assert not page.is_first_page
             assert page.is_first_page_confidence == 1
+        assert self.test_document.pages()[0].category == pred.pages()[0].category
```
Comment (zypriafl) [inline tests/trainer/test_file_splitting.py:236]: Why do we the test only the first page ("[0]")?

File: tests/trainer/test_file_splitting.py [hunk 2]
```diff
@@ -209,27 +214,26 @@ def test_splitting_no_category_document(self):
...
             assert page.is_first_page_confidence == 1
-        assert pred == test_document
+        assert pred == self.test_document
 
     def test_suggest_first_pages(self):
         """Test Splitting AI's suggesting first Pages."""
         splitting_ai = SplittingAI(self.file_splitting_model)
-        test_document = self.file_splitting_model.tokenizer.tokenize(deepcopy(self.test_document))
-        pred = splitting_ai.propose_split_documents(test_document, return_pages=True)[0]
+        pred = splitting_ai.propose_split_documents(self.test_document, return_pages=True)[0]
```
Comment (zypriafl) [inline tests/trainer/test_file_splitting.py:229]: We could improve reaibility by chaning this line to:

```
predictions = splitting_ai.propose_split_documents(self.test_document, return_pages=True)
assert len(predictions) == 1
predicition = predictions[0] 
        
   ``

File: tests/trainer/test_file_splitting.py [hunk 3]
```diff
@@ -187,6 +187,11 @@ def test_splitting_ai_evaluate_full_on_training(self):
 
     def test_splitting_ai_evaluate_full_on_testing(self):
         """Test Splitting AI's evaluate_full on testing Documents."""
+        counter = 1
+        for doc in self.file_splitting_model.test_documents:
```
Comment (zypriafl) [inline tests/trainer/test_file_splitting.py:191]: Can you add a comment explaining, why we need to set all those IDs in this test case here?

File: konfuzio_sdk/data.py [hunk 4]
```diff
@@ -435,6 +436,23 @@ def category(self) -> Optional['Category']:
         else:
             return self._category
 
+    def get_original_page(self) -> 'Page':
+        """
+        Return an original Page in case the current Page is a copy without an ID.
```
Comment (zypriafl) [inline konfuzio_sdk/data.py:441]: Please explain what "original" means here. Why do we need to get the original Page instance. Was ist different here? (=> please add this to the doctring)

File: docs/sdk/SDK and Server Integration.drawio
Comment (zypriafl) [inline docs/sdk/SDK and Server Integration.drawio:1]: Unfortunately, I cannot see the changes compared to the last version. I see a lot of "upcoming features" comments... I think they need to be updated.

image

File: tests/trainer/test_file_splitting.py (hunk 1 above)
Comment (iftwigs) [reply tests/trainer/test_file_splitting.py:236]: changed to checking all the pages

File: docs/sdk/SDK and Server Integration.drawio
Comment (iftwigs) [reply docs/sdk/SDK and Server Integration.drawio:1]: updated them

**Review by zypriafl — APPROVED** [review]

**Pull request merged**